### PR TITLE
Made Devfile part of a workspace

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Makes Devfile part of a workspace. It includes only changes of model related objects. There is not an ability to use Devfile for workspace creation yet.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13079

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: dashboard/pom.xml [hunk 1]
```diff
@@ -204,11 +204,11 @@
                                 </goals>
                                 <configuration>
                                     <target>
-                                      <!-- install dependencies -->
-                                      <exec dir="${basedir}" executable="yarn" failonerror="true">
-                                        <arg value="install" />
-                                        <arg value="--ignore-optional" />
-                                      </exec>
+                                        <!-- install dependencies -->
+                                        <exec dir="${basedir}" executable="yarn" failonerror="true">
```
Comment (skabashnyuk) [inline dashboard/pom.xml:208]: revert this change?

Comment (sleshchenko) [reply dashboard/pom.xml:208]: It's what is done by `mvn fmt:format`, so I would leave it as is to avoid reverting each time after executing of `mvn fmt:format`.

**Review by garagatyi — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Review by metlos — APPROVED** [review]

**Review by mshaposhnik — APPROVED** [review]

**Comment by sleshchenko** [comment]
ci-test
